### PR TITLE
Add libdynd and dynd-python

### PR DESCRIPTION
**Pull request by teh**
###### Things done
- [ ] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [ ] OS X
  - [x ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -14849,6 +14849,39 @@ in modules // {
...
+    # can't retrieve git version. We hardcode:
+    preConfigure = ''
+      substituteInPlace setup.py --replace "ver = check_output(['git', 'describe', '--dirty'," "ver = '${version}'"
+      substituteInPlace setup.py --replace "'--always', '--match', 'v*']).decode('ascii').strip('\n')" ""
+    '';
+
+    # Python 3 works but has a broken import test that I couldn't
+    # figure out.
+    doCheck = !isPy3k;
+    buildInputs = with pkgs; [ cmake libdynd.dev (if isPy3k then cython3 else cython) ];
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: `cython` is part of `python-packages.nix` so you can just include `self.cython`. We should get rid of all those references in `all-packages.nix`.

Comment (teh) [inline pkgs/top-level/python-packages.nix]: Ah cool, thanks! Updated.

**Comment by teh** [comment]
Hi, looks like the xcode built timed out? Cython generates very long c files that keep the compiler busy for a long time. Maybe restarting will fix it?

**Comment by FRidh** [comment]
@teh I can't restart it. And I can't test it on Darwin myself (can you?) so let's just merge and see.

**Pull request merged**